awx/__init__.py

```diff
@@ -21,6 +21,32 @@
     MODE = 'production'
 
 
+import hashlib
+from django.utils.encoding import force_bytes
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.backends.base import schema
+
+
+class FipsBaseDatabaseSchemaEditor(BaseDatabaseSchemaEditor):
+
+    @classmethod
+    def _digest(cls, *args):
+        """
+        Generates a 32-bit digest of a set of arguments that can be used to
+        shorten identifying names.
+        """
+        try:
+            h = hashlib.md5()
+        except ValueError:
+            h = hashlib.md5(usedforsecurity=False)
+        for arg in args:
+            h.update(force_bytes(arg))
+        return h.hexdigest()[:8]
+
+
```

Comment:
I'd like a django version check here similar to https://github.com/ansible/awx/blob/afe07bd87471335f97ccd324c3bb9c4adc99e17b/awx/wsgi.py#L44

Reply:
I tend to disagree, as I find maintaining that version check on Django upgrades annoying ;).  If we miss this override here on Django upgrade (and the interface changes), Tower on FIPS systems will fail catastrophically at install time (`awx-manage migrate` won't work).  As long as we have integration for FIPS, we'll catch regressions here.

Reply:
I'm more worried about the method that we are overriding changing for our non-fips mode execution. Imagine we upgrade to django 2.0 and we spend hours debugging migration errors not knowing why because the underlying implementation changes such that the _digest method is expected to return a tuple. 

Or imagine that django 2.0 accepts a patch to make _digest() behave exactly as we have overwritten it to. We will continue to carry our overwrite for no reason then. All I can do is offer up anecdotal evidence of this _happening_. https://github.com/ansible/awx/blob/devel/awx/sso/__init__.py is a monkey patch where the underlying library fixed this issue in 2016.

Reply:
https://github.com/ansible/awx/blob/devel/awx/__init__.py#L51 `find_commands` in Django 1.11.x finds `pyc` management commands now. Another example.

Reply:
Oooooookay you've convinced me :)